packages/product/src/services/product-module-service.ts

```diff
@@ -607,21 +650,41 @@ export default class ProductModuleService<
     sharedContext?: Context
   ): Promise<ProductTypes.ProductDTO[]> {
     const products = await this.create_(data, sharedContext)
-
-    return this.baseRepository_.serialize<ProductTypes.ProductDTO[]>(products, {
+    const createdProducts = await this.baseRepository_.serialize<ProductTypes.ProductDTO[]>(products, {
       populate: true,
     })
+
+    await this.eventBusModuleService_?.emit<ProductServiceTypes.ProductEventData>(
+      createdProducts.map(({ id }) => ({
+        eventName: ProductServiceTypes.ProductEvents.PRODUCT_CREATED,
+        data: { id },
+      })
+    ))
+
+    return createdProducts
   }
 
   async update(
     data: ProductTypes.UpdateProductDTO[],
     @MedusaContext() sharedContext: Context = {}
   ): Promise<ProductTypes.ProductDTO[]> {
     const products = await this.update_(data, sharedContext)
+      
+    const updatedProducts = await this.baseRepository_.serialize<ProductTypes.ProductDTO[]>(
+      products,
+      {
+        populate: true,
+      }
+    )
 
-    return this.baseRepository_.serialize<ProductTypes.ProductDTO[]>(products, {
-      populate: true,
-    })
+    await this.eventBusModuleService_?.emit<ProductServiceTypes.ProductEventData>(
```

Comment:
todo: for all emit, I am thinking that when they are done last, we don't need to wait for them, we just want the event to be emitted, so I would suggest the following approach:

```ts

try {
  return updatedProducts
} finally {
  await this.eventBusModuleService_?.emit<ProductServiceTypes.ProductEventData>
}
```

wdyt? if you agree, it needs to be changed everywhere

Reply:
I would argue that if the emit fails, the method should also throw the error.
If someone has installed the eventBus module the messages should be guaranteed.

Reply:
In which scenario the emit can fail? It will push the event to an external queue to be handled by the subscribers somewhere else. You think the queue would fail to push the event?

Reply:
True, then nevermind this comment 🤣